src/popup/Tokens/Tokens.tsx

```diff
@@ -55,43 +59,57 @@ function Tokens() {
             if (a.symbol !== 'REEF') return 1;
             return -1;
         })
-        .map((token) => (
-            <TokenCard
-                accounts={accounts as any}
-                hideBalance={false} //todo: if we want to use hiding eye
-                pools={pools}
-                tokenPrices={tokenPrices as any}
-                signer={selectedSigner as any}
-                nw={network}
-                selectedSigner={selectedSigner as any}
-                provider={provider}
-                useDexConfig={useDexConfig}
-                isReefswapUI={false} //todo: set to true if we want to show swap overlay
-                price={tokenPrices[token.address] || 0}
-                token={token}
-                tokens={tokens}
-                isDarkMode={isDarkMode}
-            />));
+        .map((token, idx) => {
+            return (
+                selectedTknPrices.data[idx].hasStatus(reefState.FeedbackStatusCode.LOADING) ? <>
+                    <Uik.Loading />
```

Comment:
display the token data progressively like in mobile app - not just after all token data is loaded - same for nft items